CHANGELOG.md

```diff
@@ -1,41 +1,64 @@
-CHANGELOG
```

Comment:
The reason I structure CHANGELOG files this way is to adhere to the suggested conventions established at http://keepachangelog.com/en/0.3.0/. It enables not only humans to read it as a markdown document, but for systems to analyze it and work with it in a structured manner. I know there isn't a stndard, but I like to adhere to this recommendation in all my packages. If you feel this is not a good direction, I will revert this back.